src/pages/index.js

```diff
@@ -38,76 +48,143 @@ const section = new Section(
+	handleNewCardSubmit
+);
 
-// POPUP: PROFILE EDIT FORM
 const profileEditFormValidator = new FormValidator(
 	configValidation,
 	profileEditForm
 );
-const profileEditPopup = new PopupWithForm(
-	selectors.modal.profileEdit.modal,
-	handleProfileFormSubmit
+
+const addNewCardFormValidator = new FormValidator(
+	configValidation,
+	addNewCardForm
 );
 
-function handleProfileFormSubmit(inputValues) {
-	const { name, about } = inputValues;
-	userInfo.setUserInfo({ name, job: about });
-	profileEditPopup.close();
-}
+const avatarEditFormValidator = new FormValidator(
+	configValidation,
+	avatarEditForm
+);
+
+const cardPreviewPopup = new PopupWithImage(selectors.modal.image.modal);
+
+// --------------------------------------------------------------//
+// Code flow--------- -------------------------------------------//
+// --------------------------------------------------------------//
+
+api.getUserInfoAndCards().then(([userData, initialCards]) => {
+	userInfo.setUserInfo(userData);
+	section.renderItems(initialCards);
+});
 
 profileEditPopup.setEventListeners();
+avatarEditPopup.setEventListeners();
+addNewCardPopup.setEventListeners();
+cardPreviewPopup.setEventListeners();
+
 profileEditButton.addEventListener(eventType.CLICK, openEditProfileModal);
+addNewCardButton.addEventListener(eventType.CLICK, openCardAddModal);
+avatarElem.addEventListener(eventType.CLICK, openProfilePicModal);
+
 profileEditFormValidator.enableValidation();
+addNewCardFormValidator.enableValidation();
+avatarEditFormValidator.enableValidation();
+
+// --------------------------------------------------------------//
+// Reusable functions -------------------------------------------//
+// --------------------------------------------------------------//
+function openCardAddModal() {
+	addNewCardFormValidator.resetValidation();
+	addNewCardPopup.open();
+}
+
+function openProfilePicModal() {
+	const { avatar } = userInfo.getUserInfo();
+	avatarInput.value = avatar;
+	avatarEditFormValidator.resetValidation();
+	avatarEditPopup.open();
+}
 
 function openEditProfileModal() {
-	const { name, job } = userInfo.getUserInfo();
+	const { name, about } = userInfo.getUserInfo();
 	profileTitleInput.value = name;
-	profileDescriptionInput.value = job;
+	profileAboutInput.value = about;
 	profileEditFormValidator.resetValidation();
 	profileEditPopup.open();
 }
 
-// POPUP: CARD ADD FORM
-const addNewCardFormValidator = new FormValidator(
-	configValidation,
-	addNewCardForm
-);
+function handleCardPreviewClick(caption, imageUrl) {
+	cardPreviewPopup.open(caption, imageUrl);
+}
 
-const addNewCardPopup = new PopupWithForm(
-	selectors.modal.cardAdd.modal,
-	handleNewCardSubmit
-);
+function handleProfileFormSubmit(inputValues) {
+	const { name, about } = inputValues;
+	const submitButtonElem = profileEditForm.querySelector(
+		selectors.modal.button
+	);
+	const oldButtonLabel = submitButtonElem.textContent;
+	submitButtonElem.textContent = 'Saving...';
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```